### PR TITLE
auto-merge and auto-approve fixes, and other improvements

### PR DESCRIPTION
**Pull request by baywet**
additional context https://github.com/std-uritemplate/std-uritemplate-csharp/actions/runs/9803350207/job/27079082638?pr=8

### CODE REVIEW

**Comment by baywet** [comment]
depends on #9 to be merged first

File: .github/workflows/dependabot-automerge-and-autoapprove.yaml [hunk 1]
```diff
@@ -14,10 +12,15 @@ jobs:
...
-        uses: dependabot/fetch-metadata@v2.1.0
+        uses: dependabot/fetch-metadata@v2.2.0
       - name: Enable auto-merge for Dependabot PRs
         if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor' }}
         run: gh pr merge --auto --squash "$PR_URL"
         env:
           PR_URL: ${{github.event.pull_request.html_url}}
-          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+      - name: Approve Dependabot PRs
```
Comment (andreaTP) [inline .github/workflows/dependabot-automerge-and-autoapprove.yaml]: The approval should run before the merge happens.

**Review by andreaTP — COMMENTED** [review]
One minor comment and lgtm, thanks for taking care of this!

File: .github/workflows/dependabot-automerge-and-autoapprove.yaml (hunk 1 above)
Comment (baywet) [reply .github/workflows/dependabot-automerge-and-autoapprove.yaml]: fixed

**Review by andreaTP — APPROVED** [review]
LGTM, thanks!

**Pull request merged**